src/librustc_mir/monomorphize/collector.rs

```diff
@@ -1166,8 +1166,13 @@ fn create_mono_items_for_default_impls<'tcx>(
                         }
                     });
 
+                    let param_env = ty::ParamEnv::reveal_all();
+                    let substs = tcx.normalize_erasing_regions(
+                        param_env,
+                        substs,
+                    );
```

Comment:
You should be able to do this on the `trait_ref`, I think? The problem appears to be that `impl_trait_ref` might return an unnormalized `TraitRef`, the per-method substs just copy from there.